Tropos/Models/DailyForecast.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+@objc class DailyForecast: NSObject {
+    let date: NSDate
+    let conditionsDescription: String
+    let highTemperature: Temperature
+    let lowTemperature: Temperature
+    
+    init(JSON: Dictionary<String, AnyObject>) {
```

Comment:
Thanks for the update. I started thinking about proper solution and I see we started thinking in similar way. 

1. Personally I don't like passing this "failable" kind of daily forecast just to indicate there is nothing to show. 
2. I like this failable initializer you proposed with class function. Let's take a look on it.

First things first. How view controller works and what RAC is doing:

View controller listen what RAC sends to it and it just iterates through its `TRDailyForecastView` views and update them. We can improve it and do this:

1. Don't create this failable class initializer which complicate things.
2. Don't create failable class which indicates that some daily forecast is not correct (we want to keep it simple so playing with additional object which is empty will complex it).
3. Implement `DailyForecast` class function that validates passed JSON and return proper object.
4. Use your `DailyForecast` initializer which looks good. We'll be always sure that object will be created because JSON validation is elswhere - simplification, Yay!

Conclusion:
- When creating `WeatherUpdate` object we're iterating from 1 to 4 and creating 3 daily forecasts. Let's change it to enumerator that will interate max 3 times but will create up to 3 objects. If something gone wrong it will end with 2 or 1 daily forecast.
- view controller will be notified via RAC that there are some forecasts to display. If there is less than 3 forecasts just hide elements starting from the last. 2 forecasts - show first 2, 1 - show one.

Solution:

    @objc class DailyForecast: NSObject {
        let date: NSDate
        let conditionsDescription: String
        let highTemperature: Temperature
        let lowTemperature: Temperature
        
        init(time: NSTimeInterval, conditionsDescription: String, highTemperature: Int, lowTemperature: Int) {
            self.date = NSDate(timeIntervalSince1970: time)
            self.conditionsDescription = conditionsDescription
            self.highTemperature = Temperature(fahrenheitValue: highTemperature)
            self.lowTemperature = Temperature(fahrenheitValue: lowTemperature)
        }
        
        class func dailyForecastWithJSON(JSON: Dictionary<String, AnyObject>) -> DailyForecast? {
            if let time = JSON["time"] as? NSTimeInterval,
                let conditionsDesc = JSON["icon"] as? String,
                let tempMax = JSON["temperatureMax"] as? Int,
                let tempMin = JSON["temperatureMin"] as? Int {
                return DailyForecast(time: time, conditionsDescription: conditionsDesc, highTemperature: tempMax, lowTemperature: tempMin)
            }
            return nil
        }
    }

I hope I didn't miss anything.